drkafka/src/main/java/com/pinterest/doctorkafka/KafkaClusterManager.java

```diff
@@ -657,10 +677,35 @@ public void handleUnderReplicatedPartitions(List<PartitionInfo> initialUrps,
    * a broker is dead or not. This method only returns true when we are sure that the broker
    * is not available.
    */
-  private boolean isDeadBroker(String host, int brokerId) {
+  private boolean isDeadBroker(String host, int brokerId, TopicPartition tp) {
     if (OperatorUtil.pingKafkaBroker(host, 9092, 5000)) {
       LOG.debug("Broker {} is alive as {}:9092 is reachable", brokerId, host);
-      return false;
+      if (tp == null) {
+        KafkaBroker broker = kafkaCluster.brokers.get(brokerId);
```

Comment:
Could you capture  line 683 - 788 into a separate method to keep the `isDeadBroker` method short?